### PR TITLE
kuksa_apps: Add support for databroker to S3 uploader

### PR DESCRIPTION
**Pull request by romainletendart**
To test this PR, make sure you `pip install -e ../../kuksa-client`.

### CODE REVIEW

File: kuksa_apps/s3/config.ini [hunk 1]
```diff
@@ -1,19 +1,38 @@
...
 
 [parquet]
 interval = 1
 max_num_rows = 10
-# seperated with ,
-path = Vehicle.Speed, Vehicle.Cabin.Infotainment.Navigation.DestinationSet.Longitude, Vehicle.Cabin.Infotainment.Navigation.DestinationSet.Latitude
-# FIXME path = Vehicle.Speed, Vehicle.Cabin.Infotainment.Navigation.DestinationSet
+# separated with ,
+paths = Vehicle.Speed, Vehicle.Cabin.Infotainment.Navigation.DestinationSet.Longitude, Vehicle.Cabin.Infotainment.Navigation.DestinationSet.Latitude
+# FIXME paths = Vehicle.Speed, Vehicle.Cabin.Infotainment.Navigation.DestinationSet
```
Comment (SebastianSchildt) [inline kuksa_apps/s3/config.ini:26]: This line is useless?

Comment (romainletendart) [reply kuksa_apps/s3/config.ini:26]: This line is related to https://github.com/eclipse/kuksa.val/issues/132.

**Review by SebastianSchildt — CHANGES_REQUESTED** [review]
The code clutters the pwed with parquet files, i.e.
```
% ls
Pipfile							kuksa_2022-Dec-12_13_39_06.734716.parquet.downloaded	kuksa_2022-Dec-12_13_40_46.032761.parquet.downloaded
config.ini						kuksa_2022-Dec-12_13_39_19.102188.parquet		kuksa_2022-Dec-12_13_40_58.423593.parquet
kuksa_2022-Dec-12_11_45_44.573104.parquet		kuksa_2022-Dec-12_13_39_19.102188.parquet.downloaded	kuksa_2022-Dec-12_13_40_58.423593.parquet.downloaded
kuksa_2022-Dec-12_13_11_08.142138.parquet		kuksa_2022-Dec-12_13_39_31.540788.parquet		kuksa_2022-Dec-12_13_41_10.865062.parquet
kuksa_2022-Dec-12_13_37_07.739068.parquet		kuksa_2022-Dec-12_13_39_31.540788.parquet.downloaded	kuksa_2022-Dec-12_13_41_10.865062.parquet.downloaded
kuksa_2022-Dec-12_13_37_07.739068.parquet.downloaded	kuksa_2022-Dec-12_13_39_43.980496.parquet		kuksa_2022-Dec-12_13_41_23.365085.parquet
kuksa_2022-Dec-12_13_37_19.232214.parquet		kuksa_2022-Dec-12_13_39_43.980496.parquet.downloaded	kuksa_2022-Dec-12_13_41_23.365085.parquet.downloaded
kuksa_2022-Dec-12_13_37_19.232214.parquet.downloaded	kuksa_2022-Dec-12_13_39_56.379624.parquet		kuksa_2022-Dec-12_13_41_35.896691.parquet
kuksa_2022-Dec-12_13_37_31.688378.parquet		kuksa_2022-Dec-12_13_39_56.379624.parquet.downloaded	kuksa_2022-Dec-12_13_41_35.896691.parquet.downloaded
kuksa_2022-Dec-12_13_37_31.688378.parquet.downloaded	kuksa_2022-Dec-12_13_40_08.817508.parquet		kuksa_2022-Dec-12_13_41_48.360673.parquet
kuksa_2022-Dec-12_13_37_44.145499.parquet		kuksa_2022-Dec-12_13_40_08.817508.parquet.downloaded	kuksa_2022-Dec-12_13_41_48.360673.parquet.downloaded
kuksa_2022-Dec-12_13_38_42.731086.parquet		kuksa_2022-Dec-12_13_40_21.293588.parquet		kuksa_2022-Dec-12_13_42_00.775607.parquet
kuksa_2022-Dec-12_13_38_42.731086.parquet.downloaded	kuksa_2022-Dec-12_13_40_21.293588.parquet.downloaded	pyarrow_mapping.py
kuksa_2022-Dec-12_13_38_54.243652.parquet		kuksa_2022-Dec-12_13_40_33.610445.parquet		readme.md
kuksa_2022-Dec-12_13_38_54.243652.parquet.downloaded	kuksa_2022-Dec-12_13_40_33.610445.parquet.downloaded	requirements.txt
kuksa_2022-Dec-12_13_39_06.734716.parquet		kuksa_2022-Dec-12_13_40_46.032761.parquet		s3_uploader.py
```

I think we want to delete files after uploading and also do not do the test download. IN function upload

 line 91  in s3_uploader.py

File: kuksa_apps/s3/config.ini (hunk 1 above)
Comment (SebastianSchildt) [reply kuksa_apps/s3/config.ini:26]: Ah I see, I suggest just removing the  "#FIXME " line to keep it simple.

**Comment by SebastianSchildt** [comment]
When data is not available(i.e. never been set) after the "Unify error handling" commit, with kuksa.val-server the resulting parquet file has NaN for unavailable data (ssems good to me), while databroker usecase breask with 404

```
kuksa_client.grpc.VSSClientError: ({'code': 404, 'reason': 'not_found', 'message': "Path 'Vehicle.Speed' not found"}, [{'path': 'Vehicle.Speed', 'error': {'code': 404, 'reason': 'not_found', 'message': "Path 'Vehicle.Speed' not found"}}, {'path': 'Vehicle.Cabin.Infotainment.Navigation.DestinationSet.Longitude', 'error': {'code': 404, 'reason': 'not_found', 'message': "Path 'Vehicle.Cabin.Infotainment.Navigation.DestinationSet.Longitude' not found"}}, {'path': 'Vehicle.Cabin.Infotainment.Navigation.DestinationSet.Latitude', 'error': {'code': 404, 'reason': 'not_found', 'message': "Path 'Vehicle.Cabin.Infotainment.Navigation.DestinationSet.Latitude' not found"}}])
```

That is not expected I think? At least not unified :)

**Comment by romainletendart** [comment]
> When data is not available(i.e. never been set) after the "Unify error handling" commit, with kuksa.val-server the resulting parquet file has NaN for unavailable data (ssems good to me), while databroker usecase breask with 404
> 
> ```
> kuksa_client.grpc.VSSClientError: ({'code': 404, 'reason': 'not_found', 'message': "Path 'Vehicle.Speed' not found"}, [{'path': 'Vehicle.Speed', 'error': {'code': 404, 'reason': 'not_found', 'message': "Path 'Vehicle.Speed' not found"}}, {'path': 'Vehicle.Cabin.Infotainment.Navigation.DestinationSet.Longitude', 'error': {'code': 404, 'reason': 'not_found', 'message': "Path 'Vehicle.Cabin.Infotainment.Navigation.DestinationSet.Longitude' not found"}}, {'path': 'Vehicle.Cabin.Infotainment.Navigation.DestinationSet.Latitude', 'error': {'code': 404, 'reason': 'not_found', 'message': "Path 'Vehicle.Cabin.Infotainment.Navigation.DestinationSet.Latitude' not found"}}])
> ```
> 
> That is not expected I think? At least not unified :)

`404-not_available` is different from `404-not_found`.
Here databroker says it doesn't know the `'Vehicle.Speed'` field.
If you provide databroker with metadata, that error will just go away.
So in my opinion, this should remain as it is.

**Pull request merged**